rewrite-core/src/main/java/org/openrewrite/trait/Reference.java

```diff
@@ -52,11 +52,13 @@ interface Provider {
 
     interface Matcher {
         boolean matchesReference(Reference value);
+        default Renamer renameTo(String newName) {
+            throw new UnsupportedOperationException();
+        }
     }
 
+    @FunctionalInterface
     interface Renamer {
-        default TreeVisitor<Tree, ExecutionContext> rename(String replacement) {
-            throw new UnsupportedOperationException();
-        }
+        String rename(String oldName);
```

Comment:
I think that perhaps extending this to `String rename(Reference reference)`  would be better here, currently we only need access to the String value, but a future Renamer might also need the value of `Kind`, or any future properties that will be added.

Reply:
Sure, let's use Reference instead. The implementation can then call getValue() on that.